subsys/bluetooth/controller/multithreading_lock.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/**
+ * @file multithreading_lock.h
+ *
+ * @brief APIs for ensuring BLE controller threadsafe operation.
+ */
+
+#ifndef MULTITHREADING_LOCK_H__
+#define MULTITHREADING_LOCK_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <zephyr.h>
+
+/** Macro for acquiring a lock */
+#define MULTITHREADING_LOCK_ACQUIRE() \
+	multithreading_lock_acquire(K_FOREVER)
+
+/** Macro for acquiring a lock without waiting. */
+#define MULTITHREADING_LOCK_ACQUIRE_NO_WAIT() \
+	multithreading_lock_acquire(K_NO_WAIT)
+
+/** Macro for acquiring a lock while waiting forever. */
+#define MULTITHREADING_LOCK_ACQUIRE_FOREVER_WAIT() \
+	multithreading_lock_acquire(K_FOREVER)
+
+/** Macro for releasing a lock */
+#define MULTITHREADING_LOCK_RELEASE() multithreading_lock_release()
+
```

Comment:
btw; this locking is used frequently with some loop that checks lock is taken which is a lot of code copying around
maybe we could create a macro to handle it - something similar to `LOCKED` from `kernel/timeout.c`?

Reply:
Post-merge?